#6 トップページにユーザ自身が作った予定一覧表示機能を実装

MaekawaShun331 opened the pull request:
・トップページにログイン/ログアウト画面へ遷移するリンクを追加しました。
・トップページに予定作成画面へ遷移するリンクを追加しました。（ログイン時のみ表示）
・トップページにユーザが作成した予定一覧を表示する機能を追加しました。（ログイン時のみ表示）
・indexのRoute処理を、indexControllerに記述するよう修正しました。

※トップページは#1 で作成したものです。

MaekawaShun331 committed:
#6 トップページにユーザ自身が作った予定一覧表示機能を実装
・#1 で作成したトップページにログイン/ログアウト画面へ遷移するリンクを追加しました。
・トップページに、ログイン時、予定作成画面へ遷移するリンクを追加しました。
・トップページに、ログイン時、ユーザが作成した予定一覧を表示する機能を追加しました。
・indexのRoute処理を、indexControllerに記述するよう修正しました。

MaekawaShun331 committed:
Merge branch 'feature/#5_ScheduleMakingPage' into feature/#6_ScheduleListDisp

jtakakura commented on app/Http/Controllers/IndexController.php:
```diff
@@ -0,0 +1,31 @@
+        $title = '予定調整くん';
+        $user = Auth::user();
+        //ログインが行われている場合、ユーザの予定表一覧を取得・表示する
+        if (!empty($user)) {
```
@MaekawaShun331 認証済みかどうかの判定は `Auth::check()` で実装してください。

https://readouble.com/laravel/5.4/ja/authentication.html

jtakakura replied:
@MaekawaShun331 `return view(...)` のところは共通化できそうなので、 `view` に渡すパラメータ部分の生成のみを分岐中に入れたらいいかと思います。

jtakakura commented on resources/views/index.blade.php:
```diff
@@ -3,4 +3,42 @@
 <h1>{{ $title }}</h1>
 <p>Welcome to {{ $title }}</p>
+  <div>
+    @empty (!$user)
```
@MaekawaShun331 認証済みかどうかの判定は上記記載の方法でお願いします。

jtakakura commented on app/Http/Controllers/IndexController.php:
```diff
@@ -0,0 +1,31 @@
+                        ->get();
+            return view('index', ['title' => $title, 'user' => $user, 'schedules' => $schedules]);
+        } else {
+            return view('index', ['title' => $title, 'user' => $user]);
```
@MaekawaShun331 ログインしていないなら、 `$user` 渡す必要がないかと思います。

jtakakura commented on app/Http/Controllers/IndexController.php:
```diff
@@ -0,0 +1,31 @@
+     */
+    public function __invoke()
+    {
+        $title = '予定調整くん';
```
@MaekawaShun331 今後どのようなタイトルが設定されるか把握できていないですが、毎回同じ `予定調整くん` を各コントローラで設定せずに済むよう実装を検討してください。

jtakakura commented on resources/views/index.blade.php:
```diff
@@ -3,4 +3,42 @@
+
+      <div>
+        <a href="/schedules/create">予定を作る</a>
+      @if (count($schedules) > 0)
```
@MaekawaShun331 修正不要ですが、 `isNotEmpty()` というメソッドもあったりします。
(Laravel の Collection の機能で、言語組み込みの array にはないですが...)

https://laravel.com/docs/5.4/collections#method-isnotempty

MaekawaShun331 committed:
#6 レビュー指摘内容修正
・認証済みかどうかの判定を'Auth::check()'で実装するよう修正しました
・indexControllerでviewに渡すパラメータを配列変数にし、処理を共通化しました
・トップページビューにて、indexControllerから渡される変数'title'を、環境設定値から取得するよう修正しました
・テンプレートビューのtitleデフォルト値がLaravelになっていたので、セキュリティを考慮し変更しました

jtakakura reviewed (approved):
LGTM
